types/mongoose-aggregate-paginate-v2/index.d.ts

```diff
@@ -0,0 +1,90 @@
+// Type definitions for mongoose-aggregate-paginate-v2 1.0
+// Project: https://github.com/webgangster/mongoose-aggregate-paginate-v2
+// Definitions by: Linus Brolin <https://github.com/linusbrolin>
+//                 simonxca <https://github.com/simonxca>
+//                 woutgg <https://github.com/woutgg>
+//                 oktapodia <https://github.com/oktapodia>
+//                 Dongjun Lee <https://github.com/ChazEpps>
+//                 gamsterX <https://github.com/gamsterx>
+//                 knyuwork <https://github.com/knyuwork>
+//                 LiRen Tu <https://github.com/tuliren>
+//                 Alexandre Croteau <https://github.com/acrilex1>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.2
+//
+// Based on type declarations for mongoose-paginate-v2 1.3.
+
+declare module 'mongoose' {
+    interface CustomLabels {
+        totalDocs?: string;
+        limit?: string;
+        page?: string;
+        totalPages?: string;
+        docs?: string;
+        nextPage?: string;
+        prevPage?: string;
+        pagingCounter?: string;
+        hasPrevPage?: string;
+        hasNextPage?: string;
+    }
+
+    interface PaginateOptions {
+        /* tslint:disable-next-line: ban-types */
+        sort?: Object | string;
```

Comment:
Why suppress this lint rule?  What's the intended type of `sort`?

Reply:
The sort is using mongoose's sorting (see https://mongoosejs.com/docs/api.html#query_Query-sort), which really specifies Obect | string as a type.

Reply:
From the readme
> Using the types Function and Object is almost never a good idea. In 99% of cases it's possible to specify a more specific type. Examples are (x: number) => number for functions and { x: number, y: number } for objects. If there is no certainty at all about the type, any is the right choice, not Object. If the only known fact about the type is that it's some object, use the type object, not Object or { [key: string]: any }

Reply:
Ok, after verification Mongoose's types define the function `sort(arg: string | any): this;`. I fixed this by specifying `any`, ignoring string because of no-any-union rule.

However, it seems like a mistake, as the header of the description (in mongoose) explains clearly what kind of object should be passed. I could therefor specify a more specific type, something like:

`{ [index: string]: 'asc' | 'desc' | 'ascending' | 'descending' | 1 | -1 }` (untested, just my best guest for the sake of example).
 
I am however against the idea of just specifying a more narrow type here than in mongoose's types, because technically mongoose could start accepting a wider range of objects without that being a breaking change. In that case, the types would be outdated.

Should I do a PR upstream and update here after it is done?

Reply:
Sorry, I'm not sure what you mean by an "upstream PR".

If `any` is too loose, you could use `object`.